docs/source/reference/changelog.md

```diff
@@ -20,6 +20,158 @@ Contributors to major version bumps in JupyterHub include:
 
 ## [Unreleased]
 
+## 5.0
+
+### 5.0.0 - 2024
+
+5.0.0 is a major release of JupyterHub.
+It has lots of cool new features.
+
+For information about upgrading, see [upgrading to 5.0 documentation](upgrading-v5).
+
+Changes that are likely to require effort to upgrade:
+
+- New JupyterHub.subdomain_hook and default subdomain scheme
+  is more reliable and should work for all usernames.
+- JupyterHub now requires Python 3.8
+- New `Authenticator.allow_all` and `allow_existing_users` configuration options
+  (the default for `allow_all` is False now, where it was True).
```

Comment:
Previously `allowed_users` being truthy led to the equivalent of `allow_existing_users=True`, and previously not having `allowed_users` or other user-allowing config have for some authenticator classes led to the equivalent of `allow_all=True`.

I perceive `(the default for `allow_all` is False now, where it was True).` to be confusing.

Reply:
tried to clear this up